### PR TITLE
Support load balancer connect to pods directly if using VPC-CNI

### PR DESCRIPTION
**Pull request by caohe**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
Most main cloud providers ([AWS](https://github.com/aws/amazon-vpc-cni-k8s), [Alibaba Cloud](https://github.com/AliyunContainerService/terway), [Tencent Cloud](https://intl.cloud.tencent.com/document/product/457/38970), etc.) support VPC-CNI now. If using VPC-CNI, pods can be attached to a load balancer directly, which reduces the cost of NodePort forwarding. However, in cloud-provider, the service controller doesn't watch Endpoints object. So changes of Endpoints won't trigger changes to the backends of the load balancer.

This PR adds watching for endpoints to the service controller. And add a start parameter as a switch to specify whether this cluster is using VPC-CNI and enables pods to be attached to a load balancer directly.
#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:
NONE
#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/cloud-provider/controllers/service/controller.go [hunk 1]
```diff
@@ -187,6 +189,20 @@ func New(
 		time.Duration(0),
 	)
 
+	if enableDirectConnect {
+		endpointsInformer.Informer().AddEventHandlerWithResyncPeriod(
```
Comment (andrewsykim) [inline staging/src/k8s.io/cloud-provider/controllers/service/controller.go:193]: service controller and the associated provider interface is designed specifically for nodes as the backend, I would recommend writing a new controller for this use-case

Comment (caohe) [reply staging/src/k8s.io/cloud-provider/controllers/service/controller.go:193]: I think a new controller is better too. do u mean an endpoints controller? it watches the endpoints, and only reconciles backends of the load balancer.

**Comment by dims** [comment]
Is this PR still needed, please rebase if so (or we can close it?)

**Comment by dims** [comment]
This PR is older than 4 weeks and needs as rebase.

Please rebase the PR against latest master and reopen if still needed.

/close

**Pull request closed without merging**